vhdbuilder/packer/configure-windows-vhd.ps1

```diff
@@ -426,13 +426,117 @@ function Update-WindowsFeatures {
     }
 }
 
+function Enable-WindowsFixInFeatureManagement {
+    Param(
+      [Parameter(Mandatory = $true)][string]
+      $Name,
+      [Parameter(Mandatory = $false)][string]
+      $Value = "1",
+      [Parameter(Mandatory = $false)][string]
+      $Type = "DWORD"
+    )
+
+    $regPath=(Get-Item -Path "HKLM:\SYSTEM\CurrentControlSet\Policies\Microsoft" -ErrorAction Ignore)
+    if (!$regPath) {
+        Write-Log "Creating HKLM:\SYSTEM\CurrentControlSet\Policies\Microsoft"
+        New-Item -Path "HKLM:\SYSTEM\CurrentControlSet\Policies\Microsoft"
+    }
+
+    $regPath=(Get-Item -Path "HKLM:\SYSTEM\CurrentControlSet\Policies\Microsoft\FeatureManagement" -ErrorAction Ignore)
+    if (!$regPath) {
+        Write-Log "Creating HKLM:\SYSTEM\CurrentControlSet\Policies\Microsoft\FeatureManagement"
+        New-Item -Path "HKLM:\SYSTEM\CurrentControlSet\Policies\Microsoft\FeatureManagement"
+    }
+
+    $regPath=(Get-Item -Path "HKLM:\SYSTEM\CurrentControlSet\Policies\Microsoft\FeatureManagement\Overrides" -ErrorAction Ignore)
+    if (!$regPath) {
+        Write-Log "Creating HKLM:\SYSTEM\CurrentControlSet\Policies\Microsoft\FeatureManagement\Overrides"
+        New-Item -Path "HKLM:\SYSTEM\CurrentControlSet\Policies\Microsoft\FeatureManagement\Overrides"
+    }
+
+    $currentValue=(Get-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Policies\Microsoft\FeatureManagement\Overrides" -Name $Name -ErrorAction Ignore)
+    if (![string]::IsNullOrEmpty($currentValue)) {
+        Write-Log "The current value of $Name in FeatureManagement\Overrides is $currentValue"
+    }
+    Set-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Policies\Microsoft\FeatureManagement\Overrides" -Name $Name -Value $Value -Type $Type
+}
+
+function Enable-WindowsFixInHnsState {
+    Param(
+      [Parameter(Mandatory = $true)][string]
+      $Name,
+      [Parameter(Mandatory = $false)][string]
```

Comment:
According to https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.management/set-itemproperty?view=powershell-7.4#-type, the content types of ``-Type`` and ``-Value`` are ``RegistryValueKind`` and ``Object``. I'm not sure whether it could cause any issue if we hardcode their types as ``string`` here.

Reply:
There is no difference between setting Name and Value via Set-ItemProperty and setting them via Enable-WindowsFixInFeatureManagement.
Set-ItemProperty will validate the input and raise an error if it is invalid.